api/payload.go

```diff
@@ -0,0 +1,103 @@
+// Copyright (c) 2016 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package api
+
+import (
+	"encoding/json"
+)
+
+// The RegisterVM payload is issued first after connecting to the proxy socket.
+// It is used to let the proxy know about a new container on the system along
```

Comment:
new container :/ the proxy doesn't really have the notion of pod as it started to support docker, so the API talks about containers and container IDs.

Maybe not ideal, but something we can review outside of this PR (we'd probably need to do changes a bit all over the place).